axum-extra/src/response/erased_json.rs

```diff
@@ -49,22 +48,24 @@ impl ErasedJson {
 
 impl IntoResponse for ErasedJson {
     fn into_response(self) -> Response {
-        let bytes = match self.0 {
-            Ok(res) => res,
-            Err(err) => {
-                return Response::builder()
-                    .status(StatusCode::INTERNAL_SERVER_ERROR)
-                    .header(header::CONTENT_TYPE, mime::TEXT_PLAIN_UTF_8.as_ref())
-                    .body(body::boxed(Full::from(err.to_string())))
-                    .unwrap();
-            }
-        };
-
-        let mut res = Response::new(body::boxed(Full::new(bytes)));
-        res.headers_mut().insert(
-            header::CONTENT_TYPE,
-            HeaderValue::from_static(mime::APPLICATION_JSON.as_ref()),
-        );
-        res
+        match self.0 {
+            Ok(bytes) => (
+                [(
+                    header::CONTENT_TYPE,
+                    HeaderValue::from_static(mime::APPLICATION_JSON.as_ref()),
+                )],
+                bytes,
+            )
+                .into_response(),
+            Err(err) => (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                [(
+                    header::CONTENT_TYPE,
+                    HeaderValue::from_static(mime::TEXT_PLAIN_UTF_8.as_ref()),
+                )],
+                err.to_string(),
```

Comment:
With `impl IntoResponse for String` already setting the `content-type` header, it is useless to set it here, no?

Reply:
Ah yes true!